hector_mapping/src/HectorMappingRos.cpp

```diff
@@ -231,106 +231,100 @@ HectorMappingRos::~HectorMappingRos()
     delete map__publish_thread_;
 }
 
-void HectorMappingRos::scanCallback(const sensor_msgs::LaserScan& scan)
-{
+void HectorMappingRos::scanCallback(const sensor_msgs::LaserScan& scan) {
   if (pause_scan_processing_) {
     return;
   }
 
-  if (hectorDrawings)
-  {
+  if (hectorDrawings) {
     hectorDrawings->setTime(scan.header.stamp);
   }
 
-  ros::WallTime startTime = ros::WallTime::now();
-
-  if (!p_use_tf_scan_transformation_)
-  {
-    if (rosLaserScanToDataContainer(scan, laserScanContainer,slamProcessor->getScaleToMap()))
-    {
+  ros::WallTime start_time = ros::WallTime::now();
+  if (!p_use_tf_scan_transformation_) {
+    // If we are not using the tf tree to find the transform between the base frame and laser frame,
+    // then just convert the laser scan to our data container and process the update based on our last
+    // pose estimate
+    if (rosLaserScanToDataContainer(scan, laserScanContainer,slamProcessor->getScaleToMap())) {
       slamProcessor->update(laserScanContainer,slamProcessor->getLastScanMatchPose());
     }
-  }
-  else
-  {
-    ros::Duration dur (0.5);
-
-    if (tf_.waitForTransform(p_base_frame_,scan.header.frame_id, scan.header.stamp,dur))
-    {
-      tf::StampedTransform laserTransform;
+  } else {
+    // If we are using the tf tree to find the transform between the base frame and laser frame,
+    // let's get that transform
+    const ros::Duration dur (0.5);
+    tf::StampedTransform laserTransform;
+    if (tf_.waitForTransform(p_base_frame_,scan.header.frame_id, scan.header.stamp,dur)) {
```

Comment:
Note how there were a lot of code inside this `if` statement after `tf_.lookupTransform`, but these were not necessary to be inside the `if` statement. I got all the code that was after `tf_.lookupTransform`, and brought it below the if/else statements (un-nesting).
Since the `else` statement has a `return`, there's no risk of reaching undesired code execution